"What's new in Svelte" June newsletter (#6356)
sveltejs/svelte: site/content/blog/2021-06-01-whats-new-in-svelte-june-2021.md

---
title: What's new in Svelte: June 2021

> description: Progress on SvelteKit V1 and tighter TypeScript/Svelte integrations in language tools

Reviewer: minor wording suggestion
```suggestion
description: Progress towards SvelteKit 1.0 and tighter TypeScript/Svelte integrations in language tools
```

author: Daniel Sandoval
authorURL: https://desandoval.net
---

> This month, we saw lots of contributions to SvelteKit and its docs. The language tools also got some new features alongside some hydration bug fixes in Svelte core. Let's jump into the updates...

Reviewer: "some hydration bug fixes in Svelte core" - I might just leave this part out here. It was a perf improvement that was added and then reverted because it was broken. I think the mention down below is fine though

> This month, we saw lots of contributions to SvelteKit and its docs. The language tools also got some new features alongside some hydration bug fixes in Svelte core. Let's jump into the updates...

Reviewer 2: ```suggestion
This month, we saw lots of contributions to SvelteKit and its docs. The language tools also got some new features, most notably deeper Svelte integration from JavaScript or TypeScript files. Let's jump into the updates...
```

> ## New in SvelteKit

Reviewer: Some of the bigger features:
* `adapter-netlify` got a fix https://github.com/sveltejs/kit/pull/1467 and new documentation in the readme https://github.com/sveltejs/kit/tree/master/packages/adapter-netlify
* support for https keypair https://github.com/sveltejs/kit/pull/1456
* now bundling Vite with SvelteKit and using an upgraded version. Remove Vite from your `package.json` if it's there
* etags for binary responses https://github.com/sveltejs/kit/pull/1382
* Renamed `$layout` to `__layout` and `$error` to `__error`
* Removed `getContext` in favor of `request.locals`
* Renamed `.svelte` output directory to `.svelte-kit`. Update your `.gitignore` accordingly

Author: Great! Split these up between "new" and "bug fixes"

> - `svelte.config.js` config files can now be loaded in ESM format (`.js` instead of `.cjs` - either can be used). It is recommended to use the `.js` format going forward.

Reviewer: > either can be used

I don't believe either can be used. I think it's forced to be `.js` and you'll get a warning or error when starting if it's not

- AMP pages will now use the rendered CSS, rather than emitted CSS
- `trailingSlash: 'never' | 'always' | 'ignore'` is now available in the config. This should make it easier to build sites that work with static hosting providers that expect a trailing slash for `index.html` pages, and provides an escape hatch for anyone that needs more complex behaviour.
- The router will no longer intercept navigation for URLs that the app does not own. This fixes a crash for apps that have `<a>` elements on the page with the same origin but don't share a base path with the app.

> - Hash only changes are now handled by the router - fixing the browser's "back" navigation between hash changes.

Reviewer: It's somewhat rare that users will encounter this fix

```suggestion
- Hash only changes are now handled by the router fixing the browser's "back" navigation between hash changes in some circumstances.
```

> - Svelte Kit builds and dev versions should now work on Windows (`pnpm build` and `dev` commands were failing due to lack of an `rm` command)

Reviewer: minor wording suggestion
```suggestion
- Svelte Kit builds and dev versions should now work on Windows (`pnpm build` and `dev` commands were failing due to usage of the `rm` command)
```

Reviewer 2: Is this even worthy to highlight here? AFAIK this is only a problem when working _on_ Kit, not _with_ Kit.

Author: Yeah probably not. I'll remove it :)

## New in Svelte & Language Tools
- Svelte 3.38.1 and 3.38.2 fixed an issue with hydration that was causing duplicate elements. If you're seeing this in your project, be sure to update the latest version!

> - You can now toggle the TypeScript plugin on and off in the Svelte extension (Extension version [**104.11.0**](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.11.0))

Reviewer 2: ```suggestion
- There now exists a TypeScript plugin which provides deeper integration with Svelte files from within JavaScript or TypeScript files. This includes diagnostics, references and renaming of variables. It is turned off by default for now. We encourage you to test it out and provide feedback  (Extension version [**104.11.0**](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.11.0))
```

> - Language server users on newer versions of Svelte will now have their sourcemaps preprocessed by Svelte instead of the (now deprecated) logic in language tools. (Extension version [**104.12.2**](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.12.2))

Reviewer 2: A new version of `svelte-check` was just released. I think this one is more important than the sourcemap thing, which not that many people would have noticed anyway.

```suggestion
- In the latest version of `svelte-check` you can now provide the path to your `tsconfig.json` or `jsconfig.json`. Example: `svelte-check --tsconfig "./tsconfig.json"`. This ensures the diagnostics are only run on files that are referenced in that config. It also runs diagnostics on JavaScript and/or TypeScript files which removes the need to run another check (like `tsc --noEmit`) for non-Svelte files (`svelte-check` version [**1.6.0**](https://github.com/sveltejs/language-tools/releases/tag/svelte-check-1.6.0))
```

- Hash only changes are now handled by the router fixing the browser's "back" navigation between hash changes in some circumstances.

## New in Svelte & Language Tools
- Svelte 3.38.1 and 3.38.2 fixed an issue with hydration that was causing duplicate elements. If you're seeing this in your project, be sure to update the latest version!
- A new TypeScript plugin provides deeper integration with Svelte files within JavaScript or TypeScript files. This includes diagnostics, references and renaming of variables. It is turned off by default for now. We encourage you to test it out and [provide feedback](https://github.com/sveltejs/language-tools/issues/580)
- In the latest version of `svelte-check` you can now provide the path to your `tsconfig.json` or `jsconfig.json`. Example: `svelte-check --tsconfig "./tsconfig.json"`. This ensures the diagnostics are only run on files that are referenced in that config. It also runs diagnostics on JavaScript and/or TypeScript files which removes the need to run another check (like `tsc --noEmit`) for non-Svelte files (`svelte-check` version [**1.6.0**](https://github.com/sveltejs/language-tools/releases/tag/svelte-check-1.6.0))

> 

Reviewer 2: Last minute update 😄 

```suggestion
- The VS Code extension and `svelte-check` got a new major release. Previously, properties that had no initializer (`export let foo;`) were only required if the user was using both TypeScript and activated `strict` mode. This is changed now: People using TypeScript, and those using `checkJs` also in JavaScript files, will now always have these properties marked as required (`svelte-check` version [**2.0.0**](https://github.com/sveltejs/language-tools/releases/tag/svelte-check-2.0.0), extension version [**105.0.0**](https://github.com/sveltejs/language-tools/releases/tag/extensions-105.0.0))

```
